src/slidedown/Slidedown.ts

```diff
@@ -17,13 +17,17 @@ import { getLoadingIndicatorWithColor } from './LoadingIndicator';
 import { getSlidedownHtml } from './SlidedownHtml';
 import { getRetryIndicator } from './RetryIndicator';
 import { SlidedownCssClasses, SlidedownCssIds, COLORS } from "./constants";
+import { Categories } from 'src/models/Tags';
+import Log from 'src/libraries/Log';
 
 export default class Slidedown {
   public options: SlidedownPermissionMessageOptions;
   public notificationIcons: NotificationIcons | null;
-  private isInSaveState: boolean;
   public isShowingFailureState: boolean;
 
+  private isInSaveState: boolean;
+  private categoryOptions?: Categories;
```

Comment:
`private categoryOptions?: Categories;` change this to be required because you always create them in constructor. then you don't need `isPrivateCategoryOptionsConfigured` check at all.

Reply:
Done.